crates/bevy_core_pipeline/src/bloom/mod.rs

```diff
@@ -32,15 +39,57 @@ use std::num::NonZeroU32;
 const BLOOM_SHADER_HANDLE: HandleUntyped =
     HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 929599476923908);
 
+const BLOOM_TEXTURE_FORMAT: TextureFormat = TextureFormat::Rg11b10Float;
+
+impl BloomSettings {
+    /// Calculates blend intensities of blur pyramid levels
+    /// during the upsampling+compositing stage.
+    ///
+    /// The function assumes all pyramid levels are upsampled and
+    /// blended into higher frequency ones using this function to
+    /// calculate blend levels every time. The final (highest frequency)
+    /// pyramid level in not blended into anything therefore this function
+    /// is not applied to it. As a result, the *mip* Earameter of 0 indicates
+    /// the second-highest frequency pyramid level (in our case that is the
+    /// 0th mip of the bloom texture with the original image being the
+    /// actual highest frequency level).
+    ///
+    /// Parameters:
+    /// * *mip* - the index of the lower frequency pyramid level (0 - max_mip, where 0 indicates highest frequency mip but not the highest frequency image).
+    /// * *max_mip* - the index of the lowest frequency pyramid level.
+    ///
+    /// This function can be visually previewed for all values of *mip* (normalized) with tweakable
+    /// BloomSettings parameters on [Desmos graphing calculator](https://www.desmos.com/calculator/ncc8xbhzzl).
+    fn compute_blend_factor(&self, mip: f32, max_mip: f32) -> f32 {
+        let x = mip / max_mip;
+
+        let mut lf_boost =
+            (1.0 - (1.0 - x).powf(1.0 / (1.0 - self.lf_boost_curvature))) * self.lf_boost;
+        let high_pass_lq =
+            1.0 - ((x - self.high_pass_frequency) / self.high_pass_frequency).clamp(0.0, 1.0);
+        // let high_pass_hq = 0.5 + 0.5 * ((x - self.high_pass_frequency) / self.high_pass_frequency).clamp(0.0, 1.0).mul(std::f32::consts::PI).cos();
```

Comment:
Can we remove these lines? Are they left commented out for a reason?

Reply:
high_pass_hq uses a cosine which results in a rounder curve but is more expensive and makes no difference to a human eye. My web visualizer uses the hq version because it matters much more there. I left it to give more clues to whoever stumbles upon that code

Reply:
Please add a comment explaining a such then :)